File: my-folder/0049-group-anagrams/solution.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        hm = defaultdict(list)
        for word in strs:
            key = "".join(sorted(word))
            hm[key].append(word)
        
        res = []
        for i in hm.values():
            res.append(i)

        return res


        
            
                


        

File: my-folder/0049-group-anagrams/solution.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {
        Map<String, List<String>> map = new HashMap<>();
        for(String str : strs){
            int[] key = new int[26];
            for(int i = 0; i < str.length(); i++){
                char curr = str.charAt(i);
                key[curr - 'a'] += 1;
            }

            String keyStr = Arrays.toString(key);
            map.computeIfAbsent(keyStr, k -> new ArrayList()).add(str);
        }

        List<List<String>> result = new ArrayList<>();
        for(List<String> val : map.values()){
            result.add(val);
        }

        return result;

    }
}

// ["eat","tea","tan","ate","nat","bat"]

